### ISSUE TITLE
Variant filter representation

### ISSUE THREAD

**Issue by nealsid** [bigdatagenomics/adam#194]
VCF format specifies that the FILT column can take on the following values:

"PASS" - variant passed all filters
Comma-separated list of failing filters - filters that the variant call failed
"." - no filter applied

(see http://gatkforums.broadinstitute.org/discussion/3401/filter-column-in-vcf-file-from-haplotypecaller for a discussion on the output of the haplotype caller)

Right now, we have two fields in VariantCallingAnnotations:

``` c++
boolean variantIsPassing
array<string> variantFilters
```

This table indicates the value of each variable for the different scenarios:

| Scenario | variantIsPassing | variantFilters |
| --- | :-: | :-: |
| Variant Passed | true | empty |
| Variant failed filter | false | failing filter list |
| No filters applied | true | empty |

@mlinderm brought up that there is some ambiguity - we don't know the difference between variant passing and no filters applied without checking another field (the list of filters applied), and also people doing analysis were only likely to check the boolean, since that's what is done with the VCF field.  The proposal is to go back to the original way, which was:

``` c++
union { null, boolean }  variantIsFiltered = null;
array<string> failingFilters = null;
```

New table:

| Scenario | variantIsFiltered | failingFilters |
| --- | :-: | :-: |
| Variant Passed | false | empty |
| Variant failed filter | true | failing filter list |
| No filters applied | null | empty |

Note that in the case where the variant passed, the passing filters should still be available from the VCF header. 

Thoughts?

**Comment by heuermh** [comment]
See also:
Adding quality, filtersFailed, and filtersPassed to Variant https://github.com/ga4gh/schemas/pull/304

**Comment by fnothaft** [comment]
Shall we close this?

**Comment by heuermh** [comment]
Fixed by #1250
